metrics/src/macros.rs

```diff
@@ -0,0 +1,870 @@
+    };
+}
+
+/// Registers a gauge.
+///
+/// Gauges represent a single value that can go up or down over time, and always starts out with an
+/// initial value of zero.
+///
+/// Metrics can be registered, which provides a handle to directly update that metric.  For gauges,
+/// [`Gauge`](crate::Gauge) is provided which can be incremented, decrement, or set to an absolute value.
+///
+/// Metric names are shown below using string literals, but they can also be owned `String` values,
+/// which includes using macros such as `format!` directly at the callsite. String literals are
+/// preferred for performance where possible.
+///
+/// # Example
+/// ```
+/// # #![no_implicit_prelude]
+/// # use ::std::string::String;
+/// # use ::std::format;
+/// # use ::std::convert::From;
+/// # use metrics::register_gauge;
+/// # fn main() {
+/// // A basic gauge:
+/// let gauge = register_gauge!("some_metric_name");
+/// gauge.increment(1.0);
+///
+/// // Specifying labels inline, including using constants for either the key or value:
+/// let gauge = register_gauge!("some_metric_name", "service" => "http");
+/// gauge.decrement(42.0);
+///
+/// const SERVICE_LABEL: &'static str = "service";
+/// const SERVICE_HTTP: &'static str = "http";
+/// let gauge = register_gauge!("some_metric_name", SERVICE_LABEL => SERVICE_HTTP);
+/// gauge.increment(3.14);
+///
+/// // We can also pass labels by giving a vector or slice of key/value pairs.  In this scenario,
+/// // a unit or description can still be passed in their respective positions:
+/// let dynamic_val = "woo";
+/// let labels = [("dynamic_key", format!("{}!", dynamic_val))];
+/// let gauge = register_gauge!("some_metric_name", &labels);
+/// gauge.set(1337.0);
+///
+/// // As mentioned in the documentation, metric names also can be owned strings, including ones
+/// // generated at the callsite via things like `format!`:
+/// let name = String::from("some_owned_metric_name");
+/// let gauge = register_gauge!(name);
+///
+/// let gauge = register_gauge!(format!("{}_via_format", "name"));
+/// # }
+/// ```
+#[macro_export]
+macro_rules! register_gauge {
+    (target: $target:expr, level: $level:expr, $name:expr $(, $label_key:expr $(=> $label_value:expr)?)* $(,)?) => {{
+        let metric_key = $crate::key_var!($name $(, $label_key $(=> $label_value)?)*);
+        let metadata = $crate::metadata_var!($target, $level);
+
+        $crate::recorder().register_gauge(&metric_key, metadata)
+    }};
+    (target: $target:expr, $name:expr $(, $label_key:expr $(=> $label_value:expr)?)* $(,)?) => {
+        $crate::register_gauge!(target: $target, level: $crate::Level::INFO, $name $(, $label_key $(=> $label_value)?)*)
+    };
+    (level: $level:expr, $name:expr $(, $label_key:expr $(=> $label_value:expr)?)* $(,)?) => {
+        $crate::register_gauge!(target: module_path!(), level: $level, $name $(, $label_key$(=> $label_value)?)*)
```

Comment:
Looks like at least for clarity's sake, this should be `$(, $label_key $(=> $label_value)?)*` instead of `$(, $label_key$(=> $label_value)?)*`. Same with the other similar callsites.

Reply:
d70bb0e6e58f01f8c5b224a98b511000db182261